code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -27,6 +27,11 @@
 	grill_loop = new(list(src), FALSE)
 	if(isnum(variant))
 		variant = rand(1,3)
+	RegisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT, .proc/on_expose_reagent)
+
+/obj/machinery/griddle/Destroy()
+	UnregisterSignal(src, COMSIG_ATOM_EXPOSE_REAGENT)
+	. = ..()
```

Comment:
Signals are unregistered when the object is destroyed, you shouldn't need this.

Reply:
oh, well, great!